### PR TITLE
Fix --output usage

### PR DESCRIPTION
**Pull request by dlespiau**
Fixes: #159

### CODE REVIEW

File: pkg/kubectl/kubectl.go [hunk 1]
```diff
@@ -34,13 +34,73 @@ type kubeCtlVersionData struct {
...
 
+// Example stdout:
+// Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", ..., Platform:"linux/amd64"}
+// Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", ..., Platform:"linux/amd64"}
+
+// We don't care about the exact reason why the parsing failed, we'll display
+// more context in the error message anyway.
+var errParsing = errors.New("parse error")
+
+func parseVersionLine(line string, versionInfo **kubeCtlVersionInfo) error {
```
Comment (rade) [inline pkg/kubectl/kubectl.go]: How closely does this match what launch-generator is doing?

Comment (dlespiau) [reply pkg/kubectl/kubectl.go]: regex ftw! https://github.com/weaveworks/launch-generator/blob/master/src/utilities/kubectl.js#L7

**Comment by dlespiau** [comment]
Note: The `lint` check is still pending as this PR renames `lint` to `global_tests`, everything passes though.

File: pkg/kubectl/kubectl.go [hunk 2]
```diff
@@ -34,13 +34,73 @@ type kubeCtlVersionData struct {
...
+		}
+		key := part[0:colon]
+
+		if key == "GitVersion" {
+			value := part[colon+2 : len(part)-1]
+			*versionInfo = &kubeCtlVersionInfo{GitVersion: value}
+		}
+	}
+
+	return nil
```
Comment (LiliC) [inline pkg/kubectl/kubectl.go]: I would prefer if this function returns the version info, rather then setting it in place. IMHO it's hard to see at first whats happening and also this way strange mutations can happen in the function and it's hard to debug. WDYT?

File: pkg/kubectl/kubectl.go [hunk 3]
```diff
@@ -49,7 +109,7 @@ func GetVersionInfo(c Client) (string, string, error) {
 			}
 			return "", "", fmt.Errorf("kubectl error (%v): %s", err, stderr)
 		}
-		return "", "", fmt.Errorf("error parsing kubectl output: %v", parseErr)
+		return "", "", fmt.Errorf("error parsing kubectl version output: %s", parseErr)
```
Comment (LiliC) [inline pkg/kubectl/kubectl.go]: Outside the scope of this PR, but would be great if we could validate and agree on a minimum required cluster and `kubectl` version that we support.

File: pkg/kubectl/kubectl.go (hunk 2 above)
Comment (dlespiau) [reply pkg/kubectl/kubectl.go]: Souds like a good idea, changed it.

Comment (leth) [reply pkg/kubectl/kubectl.go]: Did you forget to push?

Comment (dlespiau) [reply pkg/kubectl/kubectl.go]: Nope, that's the last commit in the PR (and still all green!)

File: pkg/kubectl/kubectl.go [hunk 4]
```diff
@@ -34,13 +34,75 @@ type kubeCtlVersionData struct {
...
+
 // GetVersionInfo returns the version metadata from kubectl
 // May return a value for the kubectl client version, despite also returning an error
 func GetVersionInfo(c Client) (string, string, error) {
 	// Capture stdout only (to ignore server reachability errors)
-	stdout, stderr, _, err := c.ExecuteOutputMatrix("version", "--output=json")
+	stdout, stderr, _, err := c.ExecuteOutputMatrix("version")
 	var versionData kubeCtlVersionData
-	parseErr := json.Unmarshal([]byte(stdout), &versionData)
+	parseErr := parseVersionOutput(stdout, &versionData)
```
Comment (LiliC) [inline pkg/kubectl/kubectl.go]: I would do the same here as above, for the same reasons. :) 

Make the `parseVersionOutput` take in `stdout` and return the individual versions and assign to `versionData`, rather then passing it around. Sorry for not being clearer before! (As a bonus that way we can also get rid of the double pointers ** ;) ) 

It's up to you, just a suggestion, both work! :)

Comment (dlespiau) [reply pkg/kubectl/kubectl.go]: I'm happy to change it in a subsequent PR (I'll do it straight away!) but could we ship the PR as it fixes an issue reported by users? Still need to go through the pipeline to be deployed to prod and deploying close the end of day is not a great idea :)

Comment (LiliC) [reply pkg/kubectl/kubectl.go]: SGTM, completely understandable! :)

**Comment by dlespiau** [comment]
Thanks for your understanding!

**Pull request merged**